tests/Fixer/FunctionNotation/PhpdocToReturnTypeFixerTest.php

```diff
@@ -0,0 +1,200 @@
+                '<?php /** @return int */ function my_foo(): int {}',
+                '<?php /** @return int */ function my_foo() {}',
+                null,
+                ['scalar_types' => true],
+            ],
+            'array native type' => [
+                '<?php /** @return array */ function my_foo(): array {}',
+                '<?php /** @return array */ function my_foo() {}',
+            ],
+            'callable type' => [
+                '<?php /** @return callable */ function my_foo(): callable {}',
+                '<?php /** @return callable */ function my_foo() {}',
+            ],
+            'self accessor' => [
+                '<?php
+                    class Foo {
+                        /** @return self */ function my_foo(): self {}
+                    }
+                ',
+                '<?php
+                    class Foo {
+                        /** @return self */ function my_foo() {}
+                    }
+                ',
+            ],
+            'report static as self' => [
+                '<?php
+                    class Foo {
+                        /** @return static */ function my_foo(): self {}
+                    }
+                ',
+                '<?php
+                    class Foo {
+                        /** @return static */ function my_foo() {}
+                    }
+                ',
+            ],
+            'skip resource special type' => [
+                '<?php /** @return resource */ function my_foo() {}',
+            ],
+            'skip mixed special type' => [
+                '<?php /** @return mixed */ function my_foo() {}',
+            ],
+            'null alone cannot be a return type' => [
+                '<?php /** @return null */ function my_foo() {}',
+            ],
+            'skip mixed types' => [
+                '<?php /** @return Foo|Bar */ function my_foo() {}',
+            ],
+            'nullable type' => [
+                '<?php /** @return null|Bar */ function my_foo(): ?Bar {}',
+                '<?php /** @return null|Bar */ function my_foo() {}',
+                70100,
+            ],
+            'nullable type reverse order' => [
+                '<?php /** @return Bar|null */ function my_foo(): ?Bar {}',
+                '<?php /** @return Bar|null */ function my_foo() {}',
+                70100,
+            ],
+            'nullable native type' => [
+                '<?php /** @return null|array */ function my_foo(): ?array {}',
+                '<?php /** @return null|array */ function my_foo() {}',
+                70100,
+            ],
+            'skip mixed nullable types' => [
+                '<?php /** @return null|Foo|Bar */ function my_foo() {}',
+            ],
+            'skip generics' => [
+                '<?php /** @return array<int, bool> */ function my_foo() {}',
+            ],
+            'skip array of types' => [
+                '<?php /** @return Foo[] */ function my_foo() {}',
```

Comment:
For those asking why not supporting `Foo[]` after reading https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/3640#discussion_r177971967 it's because it could also be an `Iterator`, so we can't be sure `[]` means `array`

Reply:
I think `int[]` syntax only means array of integers. [phpDocumentor's doc](https://docs.phpdoc.org/guides/types.html#arrays) and [PSR-5](https://github.com/php-fig/fig-standards/blob/master/proposed/phpdoc.md#abnf) seem to say so.

IMO if a function can return an array or an iterator, it should have something like e.g. `@return iterable` or `@return Foo[]|\Traversable`.

Reply:
`array of types` support added.